### ISSUE TITLE
Ensure CircleCI only runs "build app" step if there were changes to web-related code

### ISSUE THREAD

**Issue by madsnedergaard** [electricitymap/electricitymap-contrib#3921]
It's super annoying that capacity updates and parser changes are currently blocked by a slow CI that tests if the app can build...
We should investigate if we can split CI steps up, so it only runs the web-app-build-step if it makes sense :)

**Comment by rvankruistum** [comment]
CircleCI has a couple of options for skipping jobs. I think the easiest way would be have 2 (or more) different workflows that are executed based on branch name or commit message. What would you want to be the input that decides whether it does a full CI workflow or just a build?

I'm going to join the slack if you would rather discuss it there.

**Comment by madsnedergaard** [comment]
Hmm, I think it would be best if we can make it depend on files changed in specific folders - branch names or commit messages are too hard to remember and to enforce :)
(Lets keep discussion here on Github to keep it transparent and open)

**Comment by VIKTORVAV99** [comment]
If we are willing to switch the CI to github actions we could use this action: https://github.com/marketplace/actions/changed-files

Otherwise we could do something like this perhaps? https://gist.github.com/naesheim/18d0c0a58ee61f4674353a2f4cf71475?permalink_comment_id=2800831#gistcomment-2800831

**Comment by rvankruistum** [comment]
CircleCI has built-in functionality for this apparently: https://circleci.com/docs/2.0/configuration-cookbook/?section=examples-and-guides#execute-specific-workflows-or-steps-based-on-which-files-are-modified

I could try it out probably this week. I'll keep you posted.

**Comment by VIKTORVAV99** [comment]
@rvankruistum we discussed some related changes in #4227 and it looks like we will move towards only using GitHub Actions in the future where we already have this in place. But if you want to help migrate the cirlceci actions feel free to do so.

@madsnedergaard should we close this as we already have this functionality in the GitHub Actions CI or should we wait until we have migrated the last web related jobs?
